octavia_f5/controller/worker/tasks/f5_tasks.py

```diff
@@ -241,6 +269,34 @@ def revert(self, port: network_models.Port,
             device_response.raise_for_status()
 
 
+class GetExistingVLAN(task.Task):
+    default_provides = 'existing_vlan'
+
+    @decorators.RaisesIControlRestError()
+    def execute(self, bigip: bigip_restclient.BigIPRestClient,
+                network: f5_network_models.Network):
+        device_response = bigip.get(path=f"/mgmt/tm/net/vlan/~Common~vlan-{network.vlan_id}?expandSubcollections=true")
+        if device_response.status_code == 404:
+            return None
+        return device_response.json()
+
+
+class GetExistingRouteDomain(task.Task):
+    default_provides = 'existing_route_domain'
+
+    @decorators.RaisesIControlRestError()
+    def execute(self, bigip: bigip_restclient.BigIPRestClient,
+                network: f5_network_models.Network):
+        device_response = bigip.get(path=f"/mgmt/tm/net/route-domain/vlan-{network.vlan_id}")
+        if device_response.status_code == 404:
+            path = f"/mgmt/tm/net/route-domain/net-{network.id}"
+            device_response = bigip.get(path=path)
+
+        if device_response.status_code == 404:
+            return None
```

Comment:
Please either reraise any error (`device_response.raise_for_status()`) or remove the `RaisesIControlRestError` annotation.

Reply:
We do not need to that here, because if resource does not exist it's ok and we will create it in the next task

Reply:
In that case, my understanding is that we need to remove the `RaisesIControlRestError` decorator, since `execute` does not raise exceptions (because `raise_for_status` isn't used).

Reply:
Fixed in the commit https://github.com/sapcc/octavia-f5-provider-driver/pull/285/commits/2dae53fae0c5008a5808dc429ee55296aa5e1a1d